src/Sentry/SentryClient.cs

```diff
@@ -232,6 +230,26 @@ private SentryId DoSendEvent(SentryEvent @event, Scope? scope)
                 : SentryId.Empty;
         }
 
+        private IReadOnlyCollection<Exception>? ApplyExceptionFilters(Exception? exception)
+        {
+            var filters = _options.ExceptionFilters;
+            if (exception == null || filters == null || filters.Count == 0)
+            {
+                return null;
+            }
+
+            // Note that _options.KeepAggregateException is not relevant here.  Even if we want to keep aggregate
+            // exceptions, we would still never send one if all of its children are supposed to be filtered.
+            if (exception is AggregateException aggregate)
+            {
+                return aggregate.InnerExceptions.All(e => ApplyExceptionFilters(e) != null)
+                    ? aggregate.InnerExceptions
+                    : null;
```

Comment:
doesnt this ignore the scenario where someone has a filter explicitly for the AggregateException ?

Reply:
seems this should be 
```
    if (exception is AggregateException aggregate)
    {
        if (aggregate.InnerExceptions.All(e => ApplyExceptionFilters(e) != null))
        {
            return aggregate.InnerExceptions;
        }
    }
```

Reply:
That would cause the unfiltered case to fall through, which would invoke the filters again just to come up with the same `null` result.

In case not clear here - it's either an all or nothing.  Either all inner exceptions match the filters and the event is filtered out, or the event is not filtered out (null)

Reply:
> doesnt this ignore the scenario where someone has a filter explicitly for the AggregateException ?

Oh, I missed that actually.  That would be strange, but yeah we should handle it.  And yeah, your proposal would do it.